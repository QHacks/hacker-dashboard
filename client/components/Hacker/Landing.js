import { Container, Header, Segment } from 'semantic-ui-react';
import Emoji from 'react-emoji-render';
import React from 'react';

export default () => (
    <Container text style={{ marginTop: '3em' }}>
        <Segment raised>
            <Header as='h2'><Emoji text="Congratulations, your application is in! :tada: :heart:"/></Header>
            <p>Your application to attend QHacks 2018 has been submitted! So, what are the next steps? </p>
            <p> We will be accepting applications until January 8th 2018. Later that week we will be sending
                emails to all of the applicants updating them on their application status.</p>
            <p>If you are accepted, your next step will be to RSVP right here in the dashboard. You must
                complete this step within one week of receiving your acceptance email. The RSVP will be used
                to confirm your attendance to QHacks 2018.</p>
            <p>What if I don't get accepted? There's a chance you may be added to the waitlist. If someone
                who has been accepted fails to RSVP or can't come to the event, then we will fill their spot
                with someone from the waitlist.</p>
            <p>While you wait to hear from us there are some things you can do, such as:</p>
            <ul>
                <li>Like us on <a href="https://www.facebook.com/QHacks/">Facebook</a></li>
                <li>Follow us on <a href="https://twitter.com/QHacks18">Twitter</a></li>
                <li>Follow us on <a href="https://www.instagram.com/qhacks18/">Instagram</a></li>
            </ul>
            <p>We will be having contests for free tech on our social media so do not miss out! <Emoji
                text=":watch: :iphone:"/></p>
            <p>Finally, we want to say thank you for your excitement about QHacks! With the participation of
                people like you, we can all continue towards our goal of making Queenâ€™s and Canada as a
                whole recognized for its talent and enthusiasm for innovation and creation.</p>
        </Segment>
    </Container>
);
