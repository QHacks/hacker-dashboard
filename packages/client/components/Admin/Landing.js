import React, { Component } from "react";

export default () => <div>Admin</div>;
