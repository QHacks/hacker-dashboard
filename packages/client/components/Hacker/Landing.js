import { selectors, actionCreators } from "../../HackerStore";
import ApplicationSubmitted from "./ApplicationSubmitted";
import ApplicationWithdrawn from "./ApplicationWithdrawn";
import SuccessfulApplicant from "./SuccessfulApplicant";
import WaitlistedApplicant from "./WaitlistedApplicant";
import DeclinedApplicant from "./DeclinedApplicant";
import { Container } from "semantic-ui-react";
import React, { Component } from "react";
import { connect } from "react-redux";
import { reduce, map, merge } from "lodash";
import camelcase from "camelcase";

const { submitRSVP, withdrawApplication } = actionCreators;
const {
  getUser,
  getRSVPLoading,
  getRSVPError,
  getRSVPSubmitted,
  getWithdrawError
} = selectors;

const RSVP_FIELDS = [
  "favSnack",
  "emergencyFirstName",
  "emergencyLastName",
  "emergencyEmail",
  "emergencyPhoneNumber",
  "emergencyRelationToContact",
  "tshirtSize"
];

class HackerLanding extends Component {
  handleSubmitRSVP(values) {
    // TODO: fix this bull shit form validation
    const formComplete = RSVP_FIELDS.reduce((accum, key) => {
      if (!Object.keys(values).includes(key)) {
        return false;
      }
      return accum;
    }, true);

    if (formComplete) {
      const { user } = this.props;
      const userId = user._id;
      const eventId = user.events[0]; // TODO: Hack. Inform the dashboard about the event that is currently running
      const rsvp = reduce(
        map(values, (value, key) => {
          const emergencyContactFieldRegExp = /^emergency/;
          if (key.match(emergencyContactFieldRegExp)) {
            const newKey = camelcase(key.replace("emergency", ""));
            return {
              emergencyContact: {
                [newKey]: value
              }
            };
          }
          return { [key]: value };
        }),
        merge
      );
      this.props.submitRSVP(userId, eventId, rsvp);
    }
  }

  handleApplicationWithdraw() {
    const { user } = this.props;
    const userId = user._id;
    const eventId = user.events[0]; // TODO: Hack. Inform the dashboard about the event that is currently running
    this.props.withdrawApplication(userId, eventId);
  }

  renderCorrectStatus() {
    const { user, rsvpLoading, rsvpError } = this.props;
    const eventId = user.events[0]; // TODO: Hack. Inform the dashboard about the event that is currently running
    const application = user.applications.find(
      (application) => application.event === eventId
    );

    switch (application.status) {
    case "APPLIED":
      return <ApplicationSubmitted />;
    case "REJECTED":
      return <DeclinedApplicant />;
    case "WAITING_LIST":
      return <WaitlistedApplicant />;
    case "WITHDRAWN":
      return <ApplicationWithdrawn />;
    case "ACCEPTED":
      return (
        <SuccessfulApplicant
          rsvpStatus={application.rsvp}
          onSubmit={this.handleSubmitRSVP.bind(this)}
          onWithdrawApplication={this.handleApplicationWithdraw.bind(this)}
          rsvpLoading={rsvpLoading}
          rsvpError={rsvpError}
        />
      );
    default:
      return null;
    }
  }

  render() {
    return (
      <Container style={{ marginTop: "3em" }}>
        {this.renderCorrectStatus()}
      </Container>
    );
  }
}

function mapStateToProps(state, ownProps) {
  return {
    ...ownProps,
    user: getUser(state),
    rsvpLoading: getRSVPLoading(state),
    rsvpError: getRSVPError(state),
    rsvpSubmitted: getRSVPSubmitted(state),
    withdrawError: getWithdrawError(state)
  };
}

export default connect(mapStateToProps, { submitRSVP, withdrawApplication })(
  HackerLanding
);
