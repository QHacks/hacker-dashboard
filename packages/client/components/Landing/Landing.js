import React, { Component } from "react";

import MenuBar from "../MenuBar/index.js";
import * as constants from "../../assets/constants";

class Landing extends Component {
  render() {
    return (
      <div>
        <MenuBar wide={true} />
        <div
          css={`
            display: grid;
            @media screen and (min-width: 2500px) {
              grid-template-columns: 50% 50%;
            }
            @media screen and (max-width: 800px) {
              grid-template-columns: 100%;
              > div:last-child {
                display: none;
              }
            }
            grid-template-columns: minmax(500px, 40%) auto;
            height: 100vh;
          `}
        >
          <div
            css={`
              height: 100%;
              padding: 125px 64px 100px calc(50vw - 615px);
              @media screen and (max-width: 1400px) and (min-width: 860px) {
                padding-left: 80px;
              }
              @media screen and (max-width: 860px) {
                padding-left: 5vw;
              }
              > input {
                margin: 12px 4px !important;
                display: block;
                max-width: 375px;
              }
              p.blurb {
                max-width: 300px;
                font-size: 18px;
              }
              a.landingLink {
                font-weight: 600;
                color: ${constants.grey};
                text-decoration: underline;
              }
            `}
          >
            {this.props.children}
          </div>
          <div
            css={`
              background: url(${require("../../assets/img/circuits.png")})
                no-repeat center center;
              background-size: cover;
              display: flex;
              justify-content: center;
              align-items: center;
              height: 100%;
              border-left: 1px solid #ccd3df;
              padding-top: 84px;
            `}
          >
            <img
              src={".././assets/img/queens-building.svg"}
              css={`
                max-width: 80%;
              `}
            />
          </div>
        </div>
      </div>
    );
  }
}

export default Landing;
