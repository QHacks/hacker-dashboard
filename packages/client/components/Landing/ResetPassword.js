class ResetPassword extends Component {
  constructor(props) {
    super(props);
    this.state = {};
  }
  render() {
    return <div />;
  }
}

export default ResetPassword;
