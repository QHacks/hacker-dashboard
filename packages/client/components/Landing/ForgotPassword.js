import React, { Component } from "react";
import { Link } from "react-router-dom";
import * as constants from "../../assets/constants";
import ActionButton from "../ActionButton/ActionButton";
import Landing from "./Landing";

class ForgotPassword extends Component {
  constructor(props) {
    super(props);
    this.state = {
      email: ""
    };
  }
  render() {
    return (
      <Landing>
        <img
          src={"../../assets/img/qhacks-wordmark-colored.svg"}
          css={`
            max-height: 40px;
          `}
          alt="QHacks"
        />
        <h3
          css={`
            margin-top: 24px;
            color: ${constants.blue};
            font-weight: 700;
          `}
        >
          Forgot Password
        </h3>
        <p
          className="blurb"
          css={`
            line-height: 1.6;
            margin-top: 16px;
            color: #8a929f;
          `}
        >
          Please provide the email associated with your account to reset your
          password
        </p>
        <div
          css={`
            margin-top: 64px;
          `}
        >
          <input
            id="email"
            type="text"
            value={this.state.email}
            onChange={(e) => this.setState({ email: e.target.value })}
            placeholder="Enter your email address"
          />
        </div>
        <div
          css={`
            margin: 30px 0;
          `}
        >
          <Link className="landingLink" to="/login">
            Know you password? Login here!
          </Link>
        </div>
        <div>
          <ActionButton color="blue">Send reset link</ActionButton>
        </div>
      </Landing>
    );
  }
}

export default ForgotPassword;
