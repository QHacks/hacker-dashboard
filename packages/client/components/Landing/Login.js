import React, { Component } from "react";
import { Link } from "react-router-dom";
import { graphql } from "react-apollo";
import gql from "graphql-tag";
import axios from "axios";

import Landing from "./Landing";
import { SERVER_HOST } from "../../Client";
import ActionButton from "../ActionButton/ActionButton";
import StatusReport from "../StatusReport/StatusReport";

const ACCESS_TOKEN_STORAGE = "qhacksAccessToken";
const REFRESH_TOKEN_STORAGE = "qhacksRefreshToken";

const UPDATE_AUTHENTICATION_STATUS_MUTATION = gql`
  mutation UpdateAutheticationStatus($isAuthenticated: Boolean) {
    updateAuthenticationStatus(isAuthenticated: $isAuthenticated) @client
  }
`;

class Login extends Component {
  constructor(props) {
    super(props);

    this.state = {
      email: "",
      password: "",
      rememberMe: true
    };
  }

  async login() {
    const { email, password } = this.state;

    try {
      const response = await axios.post(`${SERVER_HOST}/oauth/session`, {
        email,
        password,
        grantType: "password"
      });

      localStorage.setItem(ACCESS_TOKEN_STORAGE, response.data.accessToken);
      localStorage.setItem(REFRESH_TOKEN_STORAGE, response.data.refreshToken);

      this.props.mutate({
        variables: {
          isAuthenticated: true
        }
      });
    } catch (err) {
      this.setState({ error: "Test Error" });
    }
  }

  render() {
    return (
      <Landing>
        <img
          src={"../../assets/img/qhacks-wordmark-colored.svg"}
          css={`
            max-height: 40px;
          `}
          alt="QHacks"
        />
        <p
          className="blurb"
          css={`
            margin-top: 24px;
            color: #8a929f;
          `}
        >
          Welcome back.
        </p>
        <p
          className="blurb"
          css={`
            margin-top: 12px;
            color: #8a929f;
          `}
        >
          Please login to your account.
        </p>
        <div
          css={`
            margin-top: 64px;
          `}
        >
          <input
            id="email"
            type="text"
            value={this.state.email}
            onChange={(e) => this.setState({ email: e.target.value })}
            placeholder="Enter your email address"
          />
          <input
            id="password"
            type="password"
            value={this.state.password}
            onChange={(e) => this.setState({ password: e.target.value })}
            placeholder="Enter your password"
          />
        </div>
        <div
          css={`
            margin: 40px 0;
            display: flex;
          `}
        >
          <div
            css={`
              flex-grow: 1;
            `}
          >
            <input
              type="checkbox"
              id="rememberMe"
              onChange={(e) => this.setState({ rememberMe: e.target.checked })}
            />
            <label htmlFor="rememberMe">Remember me</label>
          </div>
          <div
            css={`
              flex-grow: 1;
            `}
          >
            <Link className="landingLink" to="/forgot-password">
              Forgot password
            </Link>
          </div>
        </div>
        {this.state.error ? (
          <StatusReport type="danger" message={this.state.error} />
        ) : (
          ""
        )}
        <div>
          <ActionButton color="blue" onClick={() => this.login()}>
            Login
          </ActionButton>{" "}
          <ActionButton internal link="/qhacks-2019/apply">
            Apply
          </ActionButton>
        </div>
      </Landing>
    );
  }
}

export default graphql(UPDATE_AUTHENTICATION_STATUS_MUTATION)(Login);
