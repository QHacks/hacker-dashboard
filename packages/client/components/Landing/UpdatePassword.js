import React, { Component } from "react";
import { Link } from "react-router-dom";
import axios from "axios";
import ValidationError from "../ValidationError/ValidationError";
import ActionButton from "../ActionButton/ActionButton";
import { SERVER_HOST } from "../../Client";
import { blue } from "../../assets/colors";
import Alert from "../Alert/Alert";
import Landing from "./Landing";

class UpdatePassword extends Component {
  constructor(props) {
    super(props);

    this.state = {
      password: null,
      confirmPassword: null,
      alertShown: false,
      alert: {
        type: "",
        message: "",
        status: ""
      }
    };
  }

  showAlert(message, type, status = null) {
    this.setState({
      alert: {
        message,
        type,
        status
      },
      alertShown: true
    });

    setTimeout(() => this.setState({ alertShown: false }), 6000);
  }

  async updatePasswordForReset() {
    const { password } = this.state;
    const { resetHash } = this.props.match.params;

    try {
      await axios.post(`${SERVER_HOST}/oauth/updatePasswordForReset`, {
        password,
        resetHash
      });

      this.showAlert("Password has been successfully reset!", "success");
    } catch (err) {
      this.showAlert(err.response.data.message, "danger");
    }
  }

  render() {
    return (
      <Landing>
        <img
          src="../../assets/img/qhacks-wordmark-colored.svg"
          css="
            max-height: 40px;
          "
          alt="QHacks"
        />
        <h3
          css={`
            margin-top: 24px;
            color: ${blue};
            font-weight: 700;
          `}
        >
          Reset Password
        </h3>
        <p
          className="blurb"
          css="
            line-height: 1.6;
            margin-top: 16px;
            color: #8a929f;
          "
        >
          Please enter your new password.
        </p>
        <div
          css="
            margin-top: 64px;
          "
        >
          <input
            id="password"
            type="password"
            key="password"
            value={this.state.password}
            onChange={(e) => this.setState({ password: e.target.value })}
            placeholder="Enter your password"
          />
          <br />
          <ValidationError
            message={
              this.state.password !== null && this.state.password.length < 8
                ? "Password must be at least 8 characters"
                : ""
            }
          />
          <br />
          <input
            id="confirmPassword"
            type="password"
            key="confirmPassword"
            value={this.state.confirmPassword}
            onChange={(e) => this.setState({ confirmPassword: e.target.value })}
            placeholder="Confirm your password"
          />
          <br />
          <ValidationError
            message={
              this.state.confirmPassword !== null &&
              this.state.confirmPassword !== this.state.password
                ? "Passwords don't match!"
                : ""
            }
          />
        </div>
        <div
          css="
            margin: 30px 0;
          "
        >
          <Link className="landingLink" to="/login">
            Know you password? Login here!
          </Link>
        </div>
        <div>
          <ActionButton
            color="blue"
            onClick={() => this.updatePasswordForReset()}
          >
            Reset password
          </ActionButton>
        </div>
        <div
          css="
          margin: 30px 0px 0px;
        "
        >
          {this.state.alertShown ? <Alert {...this.state.alert} /> : ""}
        </div>
      </Landing>
    );
  }
}

export default UpdatePassword;
