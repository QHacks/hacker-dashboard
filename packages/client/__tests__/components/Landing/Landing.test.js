import React from "react";
import Landing from "../../../components/Landing/Landing";
import { shallow } from "enzyme";

describe("Landing", () => {
  it("renders correctly", () => {
    expect(shallow(<Landing />)).toMatchSnapshot();
  });
});
