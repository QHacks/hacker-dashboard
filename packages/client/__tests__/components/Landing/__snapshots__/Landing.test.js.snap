// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Landing renders correctly 1`] = `
<div>
  <MenuBar />
  <div
    className="css-1ixjxm2"
  >
    <div
      className="css-15c81dq"
    />
    <div
      className="css-vlu8v"
    >
      <img
        alt=""
        aria-hidden={true}
        className="css-1wzse3"
        role="none presentation"
        src=".././assets/img/queens-building.svg"
      />
    </div>
  </div>
</div>
`;
